Chrysalis-Papilio/Soen490-Papilio-Website
# PAPILIO-98-task

## General
The implementation of the Billing component was done using the window.location API. We are using the library React-router-dom for navigation. We should be consistent in our used of libraries. This is why we should change the window.location API for our default. As an added bonus, using it will clean up the code and make it easier to read.

Test for all Billing components were not done. We are aiming to write good code. Unfortunately, according to some, code without test is bad code. Writing the tests for these components will allow us to refactor the code and make change with confidence. It will also help us understand what is happening in the code. 

_issue number_: Papilio-98

## Task description
- Replace window.location with react-router-dom implementation
- Simplify links and elements passed to searchParams
- Write test for Billing elements
- Improve code for asynchronous test

## Manual Test
Make sure that you have the appropriate variables in the backend .env file. (Contact me if you don't)

1. start the backend
`npm run build && npm start`
2. start the website (it should be running to prove the npm conflict solution)
`npm start`
3. login to the dashboard
4. visit the ad page
5. Click each bundle on at a time, you should be redirect to a page for paying where the price and bundle should match what you just clicked.

## Testing
To run all the unit tests, run the following code: `npm test`
To make sure that the tests have reach the appropriate level of coverage, run the following code: `npm run coverage`
